politeiawww/cmd/piwww/testrun.go

```diff
@@ -1320,6 +1336,280 @@ func (cmd *TestRunCmd) Execute(args []string) error {
+		return err
+	}
+
+	// Create RFP
+	fmt.Println("  Create RFP")
+	np, err = newProposal(true, "")
+	if err != nil {
+		return err
+	}
+	npr, err = client.NewProposal(np)
+	if err != nil {
+		return err
+	}
+
+	// Verify RFP censorship record
+	rpr := v1.ProposalRecord{
+		Files:            np.Files,
+		Metadata:         np.Metadata,
+		PublicKey:        np.PublicKey,
+		Signature:        np.Signature,
+		CensorshipRecord: npr.CensorshipRecord,
+	}
+	err = shared.VerifyProposal(rpr, version.PubKey)
+	if err != nil {
+		return fmt.Errorf("verify RFP failed: %v", err)
+	}
+
+	// This is the RFP that we'll use for our tests
+	token = rpr.CensorshipRecord.Token
+	fmt.Printf("  RFP submitted: %v\n", token)
+
+	// Make RFP public
+	fmt.Printf("  Set RFP status: not reviewed to public\n")
+	spsc = SetProposalStatusCmd{}
+	spsc.Args.Token = token
+	spsc.Args.Status = strconv.Itoa(int(v1.PropStatusPublic))
+	err = spsc.Execute(nil)
+	if err != nil {
+		return err
+	}
+
+	pdr, err = client.ProposalDetails(spsc.Args.Token, nil)
+	if err != nil {
+		return err
+	}
+
+	if pdr.Proposal.Status != v1.PropStatusPublic {
+		return fmt.Errorf("RFP status got %v, want %v",
+			pdr.Proposal.Status, v1.PropStatusPublic)
+	}
+
+	// Authorize vote
+	fmt.Printf("  Authorize vote: authorize\n")
+	avc = AuthorizeVoteCmd{}
+	avc.Args.Token = token
+	avc.Args.Action = decredplugin.AuthVoteActionAuthorize
+	err = avc.Execute(nil)
+	if err != nil {
+		return err
+	}
+
+	// Start RFP vote
+	fmt.Printf("  Start RFP vote\n")
+	svc = StartVoteCmd{}
+	svc.Args.Token = token
+	svc.Args.Duration = 1
+	svc.Args.PassPercentage = "0"
+	svc.Args.QuorumPercentage = "0"
+	err = svc.Execute(nil)
+	if err != nil {
+		return err
+	}
+
+	// Cast RFP votes
+	fmt.Printf("  Cast RFP votes\n")
+	var noDcrwallet bool
+	vc = VoteCmd{}
+	vc.Args.Token = token
+	vc.Args.VoteID = vsr.OptionsResult[0].Option.Id
+	err = vc.Execute(nil)
+	if err != nil {
+		switch {
+		case strings.Contains(err.Error(), "connection refused"):
+			// User is not running a dcrwallet instance locally.
+			// Skipping RFP tests as we couldn't proceed without
+			// an approved RFP.
+			fmt.Printf("  WARNING: could not connect to dcrwallet; " +
+				"skipping RFP tests\n")
+			noDcrwallet = true
+
+		case strings.Contains(err.Error(), "no eligible tickets"):
+			// User doesn't have any eligible tickets. This is ok.
+			// Skipping RFP tests as we couldn't proceed without
+			// an approved RFP.
+			fmt.Printf("  WARNING: user has no elibigle tickets; " +
+				"skipping RFP tests\n")
+			noDcrwallet = true
+
+		default:
+			return err
+		}
+	}
+
+	if !noDcrwallet {
+		// Wait to RFP to finish voting
+		var vs v1.VoteSummary
+		for {
+			bvs := v1.BatchVoteSummary{
+				Tokens: []string{token},
+			}
+			bvsr, err := client.BatchVoteSummary(&bvs)
+			if err != nil {
+				return err
+			}
+
+			vs = bvsr.Summaries[token]
+			// RFP voting period has ended
+			// proceed with tests
+			if vs.Status == v1.PropVoteStatusFinished {
```

Comment:
Since we're relying on the politeiawww `--votedurationmin` to be set to something reasonable for this to not take forever, we should be validating the min vote duration.  This should be done before any of the other test run stuff starts, i.e. at the top of this `Execute` function.

Reply:
Good point, adding a validation.